tests/by-util/test_comm.rs

```diff
@@ -458,3 +463,132 @@ fn test_is_dir() {
+    let expected_stderr =
+        "comm: file 2 is not in sorted order\ncomm: input is not in sorted order\n";
+
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.write("comm1", "1\n3");
+    at.write("comm2", "3\n2");
+    scene
+        .ucmd()
+        .args(&["comm1", "comm2"])
+        .fails()
+        .code_is(1)
+        .stdout_is("1\n\t\t3\n\t2\n")
+        .stderr_is(expected_stderr);
+}
+
+#[test]
+fn test_sorted_check_order() {
+    let expected_stderr = "comm: file 2 is not in sorted order\n";
+
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.write("comm1", "1\n3");
+    at.write("comm2", "3\n2");
+    scene
+        .ucmd()
+        .arg("--check-order")
+        .args(&["comm1", "comm2"])
+        .fails()
+        .code_is(1)
+        .stdout_is("1\n\t\t3\n")
+        .stderr_is(expected_stderr);
+}
+
+#[test]
+fn test_both_inputs_out_of_order() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.write("file_a", "3\n1\n0\n");
+    at.write("file_b", "3\n2\n0\n");
+
+    scene
+        .ucmd()
+        .args(&["file_a", "file_b"])
+        .fails()
+        .code_is(1)
+        .stdout_is("\t\t3\n1\n0\n\t2\n\t0\n")
+        .stderr_is(
+            "comm: file 1 is not in sorted order\n\
+             comm: file 2 is not in sorted order\n\
+             comm: input is not in sorted order\n",
+        );
+}
+
+#[test]
+fn test_both_inputs_out_of_order_last_pair() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.write("file_a", "3\n1\n");
+    at.write("file_b", "3\n2\n");
+
+    scene
+        .ucmd()
+        .args(&["file_a", "file_b"])
+        .fails()
+        .code_is(1)
+        .stdout_is("\t\t3\n1\n\t2\n")
+        .stderr_is(
+            "comm: file 1 is not in sorted order\n\
+             comm: file 2 is not in sorted order\n\
+             comm: input is not in sorted order\n",
+        );
+}
+
+#[test]
+fn test_first_input_out_of_order_extended() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.write("file_a", "0\n3\n1\n");
+    at.write("file_b", "2\n3\n");
+
+    scene
+        .ucmd()
+        .args(&["file_a", "file_b"])
+        .fails()
+        .code_is(1)
+        .stdout_is("0\n\t2\n\t\t3\n1\n")
+        .stderr_is(
+            "comm: file 1 is not in sorted order\n\
+             comm: input is not in sorted order\n",
+        );
+}
+
+#[test]
+fn test_out_of_order_input_nocheck() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+
+    // Create input files
+    at.write("file_a", "1\n3\n");
+    at.write("file_b", "3\n2\n");
+
+    scene
+        .ucmd()
+        .arg("--nocheck-order")
+        .args(&["file_a", "file_b"])
+        .succeeds()
+        .stdout_is("1\n\t\t3\n\t2\n")
+        .no_stderr();
+}
+
+#[test]
+fn test_both_inputs_out_of_order_but_identical() {
```

Comment:
This test tests the same as the first part of `defaultcheck_order_identical_bad_order_files`.

Reply:
not exactly
test_both_inputs_out_of_order_but_identical
is with two files, defaultcheck_order_identical_bad_order_files is with the same file